test/ngAnimate/integrationSpec.js

```diff
@@ -499,6 +571,45 @@ describe('ngAnimate integration tests', function() {
         expect(child).not.toHaveClass('blue');
       });
     });
+
+    it('should cache invalid animations to avoid application of ngAnimate CSS classes', function() {
+      function fill(max) {
+        var arr = [];
+        for (var i = 0; i < max; i++) {
+          arr.push(i);
+        }
+        return arr;
+      }
+
+      inject(function($animate, $rootScope, $compile, $timeout, $$rAF, $$jqLite) {
+        ss.addRule('.animate-me', 'transition: all 0.5s;');
+
+        var classAddSpy = spyOn($$jqLite, 'addClass').andCallThrough();
+        var classRemoveSpy = spyOn($$jqLite, 'removeClass').andCallThrough();
+
+        element = jqLite(
+          '<div>' +
+            '<div ng-repeat="item in items"></div>' +
+          '</div> '
+        );
+
+        html(element);
+        $compile(element)($rootScope);
+
+        $rootScope.items = fill(100);
+        $rootScope.$digest();
+
+        expect(classAddSpy.callCount).toBe(2);
+        expect(classRemoveSpy.callCount).toBe(2);
+
+        expect(classAddSpy.calls[0].args[1]).toBe('ng-animate');
+        expect(classAddSpy.calls[1].args[1]).toBe('ng-enter');
+        expect(classRemoveSpy.calls[0].args[1]).toBe('ng-enter');
+        expect(classRemoveSpy.calls[1].args[1]).toBe('ng-animate');
+
+        expect(element.children().length).toBe(100);
+      });
+    });
```

Comment:
Can you explain exactly what you are testing here?
In what way does this test invalid animations. Actually, what do you mean by invalid animations?

My meagre understanding is that you are adding 100 items to an array. The `.animate-me` class is not applied to any of the elements in the DOM so there should be no animation, right?